src/net.cpp

```diff
@@ -740,21 +756,30 @@ bool CNode::ReceiveMsgBytes(const char* pch, unsigned int nBytes)
                 nActivityBytes += msg.hdr.nMessageSize;
 
                 // BU: furthermore, if the message is a priority message then move from the back to the front of the deque
-                // NOTE: for GET_XTHIN we don't jump the queue on test environments because the GET_XTHIN can get ahead of 
+                // NOTE: for GET_XTHIN we don't jump the queue on test environments because the GET_XTHIN can get ahead of
                 // a previous GET_XTHIN/HEADER requests and result in a DOS if the block returns out of order and with no headers
                 // in the block index or the setblockindexcandidates.
-                if ((strCommand == NetMsgType::GET_XTHIN && Params().NetworkIDString() == "main") || 
-                    strCommand == NetMsgType::XTHINBLOCK || 
-                    strCommand == NetMsgType::THINBLOCK || 
-                    strCommand == NetMsgType::XBLOCKTX || 
+                if ((strCommand == NetMsgType::GET_XTHIN && Params().NetworkIDString() == "main") ||
+                    strCommand == NetMsgType::XTHINBLOCK ||
+                    strCommand == NetMsgType::THINBLOCK ||
+                    strCommand == NetMsgType::XBLOCKTX ||
                     strCommand == NetMsgType::GET_XBLOCKTX )
                 {
                     vRecvMsg.push_front(msg);
                     vRecvMsg.pop_back();
                     LogPrint("thin", "Receive Queue: pushed %s to the front of the queue\n", strCommand);
                 }
-           }
+            }
             // BU: end
+
+            //store received bytes per message command
+            //to prevent a memory DOS, only allow valid commands
+            mapMsgCmdSize::iterator i = mapRecvBytesPerMsgCmd.find(msg.hdr.pchCommand);
+            if (i == mapRecvBytesPerMsgCmd.end())
+                i = mapRecvBytesPerMsgCmd.find(NET_MESSAGE_COMMAND_OTHER);
+            assert(i != mapRecvBytesPerMsgCmd.end());
```

Comment:
I'm not sure of the context here, but is the input sanitized somehow before getting to that point ? Because if not, you know where this is going.

It's a good idea to wrap some data into a type once it has passed validation so that the type checker can ensure proper validation is done.